### PR TITLE
fix(AP-1778): handle falsy values in context.get()

### PR DESCRIPTION
**Pull request by mattstrom**
* add unsubscribe for promise.listen()
* introduce type declarations

### CODE REVIEW

File: src/ponyfills/objects.js [hunk 1]
```diff
@@ -81,6 +81,25 @@ export function removeValueForKey(key, map) {
   return recurse(key.split('.'), 0, map);
 }
 
+export function hasKey(key, map) {
```
Comment (robertsevernsentient) [inline src/ponyfills/objects.js:84]: Test needed

File: src/types.d.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  variants: { active: string[] };
+  experiments: {
+    allocations: Allocation[];
+    confirmations: Confirmation[];
+    exclusions: any[];
+  };
+  device: string;
+  location: string;
+  platform: string;
+  geo: {
```
Comment (robertsevernsentient) [inline src/types.d.ts:118]: Should these be part of the interface? They are dependent on the info we send back from configuration. Do we explicitly set them in the javascript-sdk codebase?

**Review by robertsevernsentient — APPROVED** [review]

File: src/types.d.ts (hunk 2 above)
Comment (mattstrom) [reply src/types.d.ts:118]: Hmm... Explicitly no. They are spliced in from _configuration.json_.

Comment (mattstrom) [reply src/types.d.ts:112]: I split out the context from PAPI into a separate interface that `RemoteContext` extends from. I have another PR inbound that depends on this, so I'm going to merge this one. I can remove the interface in that PR if you still feel uneasy about it.

**Pull request merged**